### PR TITLE
61455 add component aux resources

### PR DESCRIPTION
**Pull request by nilsgstrabo**
- Show OAuth service replicas in component page
- Restart OAuth service
- OAuth replica log
- Extract start, stop and restart state, reducer and sagas as reusable code
- Extract replica log to reusable component

Navigate to application with name 'radix-oauth2-example' for usage

### CODE REVIEW

File: src/components/page-active-component/oauth-toolbar.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    return (
+      <div className="component-actions">
+        <Button
+          onClick={this.doRestartOAuthService}
+          disabled={!isRestartEnabled}
+          variant="outlined"
+        >
+          Restart
+        </Button>
+        {restartInProgress && <CircularProgress size="32" />}
```
Comment (Pespiri) [inline src/components/page-active-component/oauth-toolbar.js:56]: size expects input as number, not string
```suggestion
        {restartInProgress && <CircularProgress size={32} />}
```

File: src/components/page-active-component/component-replica-list.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    getReplicaUrl(appName, envName, componentName, replicaName);
+
+export const ComponentReplicaList = ({
+  appName,
+  envName,
+  componentName,
+  replicaList,
+}) => {
+  return (
+    <React.Fragment>
```
Comment (Pespiri) [inline src/components/page-active-component/component-replica-list.js:19]: Omit the optional React.Fragment inside of empty tags
```suggestion
    <>
```

File: src/components/page-oauth-replica/index.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        />
+      </AsyncResource>
+    </>
+  );
+};
+
+PageOAuthAuxiliaryReplica.propTypes = {
+  appName: PropTypes.string.isRequired,
+  componentName: PropTypes.string.isRequired,
+  deploymentName: PropTypes.string,
```
Comment (Pespiri) [inline src/components/page-oauth-replica/index.js:77]: remove unused field
```suggestion
```

File: src/components/replica/index.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            <Typography>Restarted {replica.restartCount} times</Typography>
+          </div>
+        )}
+      </>
+    )}
+  </>
+);
+
+export const Replica = (props) => {
+  const { logState, replica, title, duration, status, state } = props;
```
Comment (Pespiri) [inline src/components/replica/index.js:63]: add all expected fields for input props to `Replica.propTypes`, not just `replica`

**Review by Pespiri — CHANGES_REQUESTED** [review]

File: src/components/page-active-component/component-replica-list.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import PropTypes from 'prop-types';
+import ReplicaSummaryModel from '../../models/replica-summary';
+import React from 'react';
+import { Typography } from '@equinor/eds-core-react';
+import { getReplicaUrl } from '../../utils/routing';
+import { ReplicaList } from '../replica-list';
+
+const replicatUrlFuncFactory =
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply src/components/page-active-component/component-replica-list.js:8]: Yes :) Fixed

File: src/components/page-active-component/component-replica-list.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+      )}
+    </React.Fragment>
+  );
+};
+
+ComponentReplicaList.propTypes = {
+  appName: PropTypes.string.isRequired,
+  envName: PropTypes.string.isRequired,
+  componentName: PropTypes.string.isRequired,
+  replicaList: PropTypes.arrayOf(PropTypes.exact(ReplicaSummaryModel)),
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply src/components/page-active-component/component-replica-list.js:43]: Array can be undefined empty, but I agree that it is probably better to use shape. Fixed.

File: src/components/page-active-component/oauth-service.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+                <StatusBadge type={oauth2.deployment.status}>
+                  {oauth2.deployment.status}
+                </StatusBadge>
+              </div>
+            )}
+          </div>
+          <div className="grid grid--table-overflow">
+            {oauth2.deployment &&
+            oauth2.deployment.replicaList &&
+            oauth2.deployment.replicaList.length > 0 ? (
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply src/components/page-active-component/oauth-service.js:41]: fixed

File: src/components/page-active-component/oauth-service.js [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+import { OAuthAuxiliaryResourceModel } from '../../models/oauth-auxiliary-resource';
+import OAuthToolbar from './oauth-toolbar';
+
+const replicatUrlFuncFactory =
+  (appName, envName, componentName) => (replicaName) =>
+    getOAuthReplicaUrl(appName, envName, componentName, replicaName);
+
+export const OAuthService = ({ appName, envName, componentName, oauth2 }) => {
+  return (
+    <React.Fragment>
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply src/components/page-active-component/oauth-service.js:16]: fixed

File: src/components/page-active-component/oauth-toolbar.js [hunk 9]
```diff
@@ -0,0 +1,83 @@
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import { oauthAuxiliaryResourceRestartState } from '../../state/oauth-auxiliary-resource';
+import componentStatuses from '../../state/component/component-states';
+import oauthActions from '../../state/oauth-auxiliary-resource/action-creators';
+import requestStatuses from '../../state/state-utils/request-states';
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply src/components/page-active-component/oauth-toolbar.js:8]: fixed

File: src/components/page-active-component/oauth-toolbar.js [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+      oauth2.deployment &&
+      oauth2.deployment.status === componentStatuses.CONSISTENT &&
+      oauth2.deployment.replicaList != null &&
+      oauth2.deployment.replicaList.length > 0 &&
+      restartRequestStatus !== requestStatuses.IN_PROGRESS;
+
+    const restartInProgress =
+      restartRequestStatus === requestStatuses.IN_PROGRESS ||
+      (oauth2 &&
+        oauth2.deployment &&
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply src/components/page-active-component/oauth-toolbar.js:43]: fixed

File: src/components/page-active-component/oauth-toolbar.js [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+
+  render() {
+    const { oauth2, restartRequestStatus, restartRequestMessage } = this.props;
+
+    const isRestartEnabled =
+      oauth2 &&
+      oauth2.deployment &&
+      oauth2.deployment.status === componentStatuses.CONSISTENT &&
+      oauth2.deployment.replicaList != null &&
+      oauth2.deployment.replicaList.length > 0 &&
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply src/components/page-active-component/oauth-toolbar.js:37]: fixed

File: src/components/page-environment/replicas.js [hunk 12]
```diff
@@ -5,7 +5,7 @@ import { Typography } from '@equinor/eds-core-react';
 
 const Replicas = ({ appName, envName, componentName, replicaList }) => (
   <>
-    {replicaList ? (
+    {replicaList && replicaList.length > 0 ? (
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply src/components/page-environment/replicas.js:8]: fixed

File: src/components/page-oauth-replica/use-select-replica.js [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+export const useSelectReplica = (environment, componentName, replicaName) => {
+  const [replica, setReplica] = useState();
+
+  useEffect(() => {
+    const deployment = environment ? environment.activeDeployment : null;
+
+    const component =
+      deployment && deployment.components
+        ? deployment.components.find((comp) => comp.name === componentName)
+        : null;
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply src/components/page-oauth-replica/use-select-replica.js:13]: fixed

File: src/components/page-oauth-replica/use-select-replica.js [hunk 14]
```diff
@@ -0,0 +1,28 @@
+import { useState, useEffect } from 'react';
+import replicaSummaryNormaliser from '../../models/replica-summary/normaliser';
+
+export const useSelectReplica = (environment, componentName, replicaName) => {
+  const [replica, setReplica] = useState();
+
+  useEffect(() => {
+    const deployment = environment ? environment.activeDeployment : null;
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply src/components/page-oauth-replica/use-select-replica.js:8]: fixed

File: src/components/page-oauth-replica/use-select-replica.js [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+
+    const selectedReplica =
+      component &&
+      component.oauth2 &&
+      component.oauth2.deployment &&
+      component.oauth2.deployment.replicaList
+        ? component.oauth2.deployment.replicaList.find(
+            (replica) => replica.name === replicaName
+          )
+        : null;
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply src/components/page-oauth-replica/use-select-replica.js:23]: fixed

File: src/components/page-active-component/active-component-overview.js [hunk 16]
```diff
@@ -23,7 +24,6 @@ import {
 } from '../../state/subscriptions/action-creators';
 import { getEnvsUrl } from '../../utils/routing';
 import { routeWithParams } from '../../utils/string';
-
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply src/components/page-active-component/active-component-overview.js:26]: fixed

File: src/components/page-active-component/oauth-toolbar.js (hunk 1 above)
Comment (nilsgstrabo) [reply src/components/page-active-component/oauth-toolbar.js:56]: fixed

File: src/components/page-active-component/oauth-service.js [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+              <ReplicaList
+                replicaList={oauth2.deployment.replicaList}
+                replicaUrlFunc={replicatUrlFuncFactory(
+                  appName,
+                  envName,
+                  componentName
+                )}
+              />
+            ) : (
+              <Typography variant="body_short">
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply src/components/page-active-component/oauth-service.js:51]: fixed

File: src/components/page-active-component/component-replica-list.js [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+        <ReplicaList
+          replicaList={replicaList}
+          replicaUrlFunc={replicatUrlFuncFactory(
+            appName,
+            envName,
+            componentName
+          )}
+        />
+      ) : (
+        <Typography variant="body_short">
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply src/components/page-active-component/component-replica-list.js:31]: fixed

File: src/components/page-active-component/component-replica-list.js (hunk 2 above)
Comment (nilsgstrabo) [reply src/components/page-active-component/component-replica-list.js:19]: fixed

File: src/components/page-oauth-replica/index.js (hunk 3 above)
Comment (nilsgstrabo) [reply src/components/page-oauth-replica/index.js:77]: fixed

File: src/components/replica/index.js [hunk 19]
```diff
@@ -0,0 +1,113 @@
...
+            <Typography>This replica has no log</Typography>
+          )}
+        </AsyncResource>
+      </section>
+    </>
+  );
+};
+
+Replica.propTypes = {
+  replica: PropTypes.exact(ReplicaSummaryModel),
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply src/components/replica/index.js:112]: fixed

File: src/components/replica/index.js (hunk 4 above)
Comment (nilsgstrabo) [reply src/components/replica/index.js:63]: fixed

**Review by Pespiri — APPROVED** [review]

**Pull request merged**